R/vad.R

```diff
@@ -0,0 +1,183 @@
+#' @param vp A vertical profile to annotate the VAD plot with the fit in the vertical profile
+#' @param range The distance range in to filter the range gates with.
+#'      If a `vp` is provided the range is taken from there and the argument `range` should not be provided.
+#' @param height The height range to filter the range gates by.
+#'      If only one value is provided next to a `vp` then the height bin intersection with this elevation is plotted.
+#' @param range_gate_filter Optional filtering of the range gates. By default range gates are filtered for a `DBZH` less then 20 and a `RHOHV` less then 0.95.
+#'      Alternative filters could be used to highlight specific effects.
+#' @param plotting_geom The geom function to visualize the range gates, the default is [ggplot2::geom_point()], in some cases this suffers from over plotting.
+#'      Alternatives that avoid over plotting could be [ggplot2::geom_bin2d()] or [ggpointdensity::geom_pointdensity()].
+#' @param plotting_geom_args A list with additional arguments to the `plotting_geom` function. For example, controlling the point size or alpha.
+#' @param annotate A [glue][glue::glue()] string that is used to annotate the plot with additional properties of the height bin of the `vp`.
+#'      The string is evaluated using the columns from `as.data.frame(vp)`.
+#'      Use `NULL` if no annotation is desired.
+#' @param vp_color The color used for the `vp` annotations and line.
+#' @param annotation_size The text size used for the annotation.
+#'
+#' @returns A [ggplot2::ggplot] object.
+#'
+#' @details  As a [ggplot2::ggplot] object  us returned additional elements can be added using regular function in
+#'      the `ggplot2` package. Labels could, for example, be modified using [ggplot2::labs()] or [ggplot2::ggtitle()].
+#'      Using [ggplot2::theme()] the visual appearance can easily be modified. To do this the regular [+][ggplot2::+.gg]
+#'      syntax can be used. In the examples this is demonstrated using scale.
+#'
+#' @export
+#' @examples
+#' pvolfile <- system.file("extdata", "volume.h5", package = "bioRad")
+#' example_pvol <- read_pvolfile(pvolfile)
+#' # VAD plots can be created for polar volumes alone
+#' vad(example_pvol,
+#'   range = c(5000, 30000), height = c(200, 400)
+#' )
+#' # It is also possible to plot one height or more height bins from a `vp`.
+#' # Many visual aspects can be controlled through the function arguments
+#' # or through adding `ggplot`
+#' vp <- calculate_vp(pvolfile)
+#' vad(example_pvol,
+#'   vp = vp,
+#'   height = 400,
+#'   plotting_geom_args=list(ggplot2::aes(color=ZDR))
+#' ) + ggplot2::scale_color_gradient2()
+#' vad(example_pvol,
+#'   vp = vp, height=c(400,1200),
+#'   plotting_geom=ggplot2::geom_bin2d,
+#'   annotate="sdvvp: {round(sd_vvp,2)} [m/s]",
+#' ) + ggplot2::scale_fill_viridis_c()
+vad <- function(x, ...) {
+  UseMethod("vad", x)
+}
+#' @rdname vad
+#' @export
+vad.pvol <- function(x, vp = NULL,..., range = NULL, height = NULL,
```

Comment:
In general there are very little inline comments, it would be helpful to have more text added to understand what happens in each code block in the function